targets/lenovo-x1/everything.nix

```diff
@@ -143,6 +156,33 @@
               };
             };
           })
+
+          #TODO: how to handle the majority of laptops that need a little
+          # something extra?
+          # SEE: https://github.com/NixOS/nixos-hardware/blob/master/flake.nix
+          # nixos-hardware.nixosModules.lenovo-thinkpad-x1-10th-gen
+
+          ({config, ...}: {
+            boot.kernelParams = let
+              filterDevices = builtins.filter (d: d.vendorId != null && d.productId != null);
+              mapPciIdsToString = builtins.map (d: "${d.vendorId}:${d.productId}");
+              vfioPciIds = mapPciIdsToString (filterDevices (
+                config.ghaf.hardware.definition.network.pciDevices
+                ++ config.ghaf.hardware.definition.gpu.pciDevices
+                ++ config.ghaf.hardware.definition.audio.pciDevices
+              ));
+            in [
+              "intel_iommu=on,sm_on"
+              "iommu=pt"
+              # Prevent i915 module from being accidentally used by host
+              "module_blacklist=i915"
+              "acpi_backlight=vendor"
+
+              "vfio-pci.ids=${builtins.concatStringsSep "," vfioPciIds}"
+            ];
+
+            boot.initrd.availableKernelModules = ["nvme"];
+          })
```

Comment:
there is something wrong here as this has all been removed and refactored to linwa 42-53 above